docs/api/matching/ndt.md

```diff
@@ -0,0 +1,66 @@
+# wave/matching/ndt.hpp
+
+This is a wrap of NDT in PCL
+
+## Type Definitions
+
+    namespace wave {
+
+    typedef pcl::PointCloud<pcl::PointXYZ>::Ptr PCLPointCloud;
+
+    }  // end of wave namespace
+
+This type definition is used as a shorthand for the pointcloud object type used
+by the scan matching implementations in PCL. A number of those implementations are
+wrapped, so they use the same datatype for the pointcloud.
+
+## ICP Matcher
+
+    class NDTMatcher : public Matcher<PCLPointCloud> {
+     public:
+        explicit NDTMatcher(float resolution, const std::string &config_path);
+        void setRef(const PCLPointCloud &ref);
+        void setTarget(const PCLPointCloud &target);
+        bool match();
+
+     private:
+        pcl::NormalDistributionsTransform<pcl::PointXYZ, pcl::PointXYZ> ndt;
+        pcl::ApproximateVoxelGrid<pcl::PointXYZ> filter;
+        PCLPointCloud ref, target, final;
+        const float min_res = 0.05f;
```

Comment:
yes, its more explicit, without it C++ treats it as a double and will attempt to cast double to float.